src/pages/library/DataExplorer.js

```diff
@@ -37,21 +114,58 @@ const DataExplorer = props => {
             // Don't call Nav.history.replace(). That will trigger a request and
             // cause the page to flicker.
             const url = window.location.origin + '#' + Nav.history.location.pathname.slice(1) + '?' + message.deQueryStr
-            window.history.replaceState({}, 'Data Explorer - ' + props.dataset, url)
+            window.history.replaceState({}, 'Data Explorer - ' + dataset, url)
           }
         }
       }
     })
-  ])
-}
 
+    const paragraphStyle = { style: { margin: '2rem' } }
+    const notInAuthDomainError = div({
+      style: {
+        fontSize: 18,
+        margin: '3rem',
+        width: 800
+      }
+    }, [
+      div(paragraphStyle, ['This Data Explorer requires you to be in the ',
+        span({ style: { fontWeight: 'bold' } }, this.authDomain),
+        ' Terra group.']),
+      div(paragraphStyle, 'If you have a different Google account in that ' +
+        'group, please sign out of Terra and sign in with that account. To ' +
+        'sign out of Terra, click on the menu on the upper left, ' +
+        'click on your name, then click Sign Out'),
+      div(paragraphStyle, [
+        'If you don\'t have a Google account in that group, please ',
+        link({ onClick: () => { contactUsActive.set(true) } }, 'apply for access.')
+      ])
+    ])
+
+    return h(Fragment, [
+      header,
+      Utils.cond(
+        [this.authDomain === undefined, dataExplorer],
```

Comment:
Since the majority of this logic is only relevant to the private explorers, it might be cleaner to modularize things a bit. E.g. keep the old `DataExplorer` component largely the same, continue to use that directly for the public explorer pages, and create a wrapper component for the private pages that does the various auth checks, and finally renders the `DataExplorer` once everything is ready.